### PR TITLE
WIP: Package and User Area Redesign

### PR DESCRIPTION
**Pull request by designxsteph**
Contains redesign of Packages area and Login area. Some pages still need to be converted to the new design within these sections. Optimization needs to be implemented on new css libraries by either extending cassette or implementing gulp.

### CODE REVIEW

File: chocolatey/Website/Views/Packages/ContactOwners.cshtml [hunk 1]
```diff
@@ -1,29 +1,32 @@
 ﻿@using PoliteCaptcha
 @model ContactOwnersViewModel
 @{
-  ViewBag.Tab = "Packages";
-  Bundles.Reference("Scripts/validation");
+    ViewBag.Tab = "Packages";
+    Layout = "../Shared/OldLayout.cshtml"; //Delete this when converting to new layout
```
Comment (ferventcoder) [inline chocolatey/Website/Views/Packages/ContactOwners.cshtml]: curious?

By the way, pathing for layouts - `~/Shared/OldLayout.cshtml` (the `~` being a site root thing that is available in ASP.NET MVC)

File: chocolatey/Website/Views/Packages/Delete.cshtml [hunk 2]
```diff
@@ -1,71 +1,70 @@
...
-<p>
-    Unlisting a package hides the package from search results and all NuGet commands, but packages 
-    are still available for download. For example, they can still be downloaded as dependencies to 
-    other packages.
-</p>
- @if (!Model.Listed && Model.Status != PackageStatusType.Approved && Model.Status != PackageStatusType.Exempted)
+﻿@model DisplayPackageViewModel
+@{
+    ViewBag.Tab = "Packages";
+    Layout = "../Shared/OldLayout.cshtml"; //Delete this when switched to new PackagesLayout
```
Comment (ferventcoder) [inline chocolatey/Website/Views/Packages/Delete.cshtml]: Same comment as above

File: chocolatey/Website/Views/Packages/Edit.cshtml [hunk 3]
```diff
@@ -1,16 +1,17 @@
...
-    NuGet currently does not allow updating package metadata on the website. This helps ensure 
-    that the package itself (and the source used to build the package) remains the one true 
-    source of package metadata.
-</p>
-<p>
-    This does require that you increment the package version.
-</p>
+﻿@{
+    ViewBag.Tab = "Packages";
+    Layout = "../Shared/OldLayout.cshtml"; //Delete this when switched to new PackagesLayout
```
Comment (ferventcoder) [inline chocolatey/Website/Views/Packages/Edit.cshtml]: Same note on layout links. Don't use `../` anywhere if you can help it.

File: chocolatey/Website/Views/Packages/DisplayPackage.cshtml [hunk 4]
```diff
@@ -79,910 +87,1080 @@
...
-        var nuspecId = @Model.Id.Replace(".","-") + "-nuspec";
-        <ul>
-          <li>@Model.Id<span>.nuspec</span> 
-            <a onclick="if ($(this).text() == 'Hide') {$('#@nuspecId').hide();$(this).text('Show');} else {$('#@nuspecId').show();$(this).text('Hide');}">Hide</a>
-            <br />
-            <div id="@nuspecId" class="package-file-content">    
-              <pre class="brush: xml">
-&lt;?xml version="1.0" encoding="utf-8"?&gt;
-&lt;!-- This is not exact, it's only based on what is stored about the package --&gt;
-&lt;!-- description and releaseNotes left ou
```
Comment (ferventcoder) [inline chocolatey/Website/Views/Packages/DisplayPackage.cshtml]: This page is going to be really interesting to view - the diff is really hard to follow...

File: chocolatey/Website/Views/Packages/UploadPackage.cshtml [hunk 5]
```diff
@@ -1,25 +1,25 @@
...
-        <div class="form-field">
-            <label for="PackageFile">Choose a package...</label>
-            <input type="file" name="UploadFile" />
-        </div>
-
-        <input type="submit" value="Upload" title="Upload the package." />
-    </fieldset>
+﻿@using NuGetGallery
+@{
+    Layout = "../Shared/OldLayout.cshtml"; //Delete this section entirely when switching to new PackageLayout, will automatically pull new template
```
Comment (ferventcoder) [inline chocolatey/Website/Views/Packages/UploadPackage.cshtml]: Just so you don't lose track of these 👍

File: chocolatey/Website/Views/Search/SearchResults.cshtml [hunk 6]
```diff
@@ -1,104 +1,136 @@
...
-            gcse.type = 'text/javascript';
-            gcse.async = true;
-            gcse.src = 'https://cse.google.com/cse.js?cx=' + cx;
-            var s = document.getElementsByTagName('script')[0];
-            s.parentNode.insertBefore(gcse, s);
-        })();
-</script>
-}
+﻿@{
+    Layout = "~/Views/Shared/PackagesLayout.cshtml";
```
Comment (ferventcoder) [inline chocolatey/Website/Views/Search/SearchResults.cshtml]: 👍

File: chocolatey/Website/Views/Packages/Edit.cshtml (hunk 3 above)
Comment (designxsteph) [reply chocolatey/Website/Views/Packages/Edit.cshtml]: These paths have been updated here: https://github.com/chocolatey/chocolatey.org/pull/611/commits/e8c5c5d0103528094a9c99e6c794da25b3990008

**Comment by gep13** [comment]
@ferventcoder could we push this to `chocotest` so that we can see it in action, and take it for a spin?

**Comment by ferventcoder** [comment]
I think we can push it up there.

File: chocolatey/Website/Scripts/packages/package-details.js [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+    $this.toggleClass('btn-dark');
+    if ($this.hasClass('btn-dark')) {
+        $this.html('Hide').removeClass('btn-secondary');
+    } else {
+        $this.html('Show').removeClass('btn-dark').addClass('btn-secondary');
+    }
+});
+// If expanded for moderation
+var $files = $('.moderation-view .collapse-2');
+var $expand = $(".moderation-view .file-name:contains('chocolateyInstall.ps1'), .moderation-view .file-name:contains('chocolateyinstall.ps1'), .moderation-view .file-name:contains('chocolateyUninstall.ps1'), .moderation-view .file-name:contains('chocolateyuninstall.ps1'), .moderation-view .file-name:contains('chocolateyBeforeModify.ps1'), .moderation-view .file-name:contains('chocolateybeforemodify.ps1'), .moderation-view .file-name:contains('vertification.txt'), .moderation-view .file-name:contains('VERTIFICATION.txt'), .moderation-view .file-name:contains('verification.txt'), .moderation-view .file-name:contains('VERIFICATION.txt'), .moderation-view .file-name:contains('license.txt'), .moderation-view .file-name:contains('LICENSE.txt')");
```
[the remark this answers is not in the document]
Comment (designxsteph) [reply chocolatey/Website/Scripts/packages/package-details.js]: This has been added here https://github.com/chocolatey/chocolatey.org/pull/611/commits/de155fce1b7f359dbdced7637b1e4f9f8eb98b46

File: chocolatey/Website/Scripts/packages/package-details.js [hunk 8]
```diff
@@ -0,0 +1,153 @@
...
+            $h4.text(textParts[0]).after('<h6 class="comment-date">on ' + textParts[1] + '</h6>');
+        }
+    });
+
+    // Wrap in li
+    $('.user-role').each(function () {
+        $(this).nextUntil('.user-role').addBack().wrapAll('<li></li>');
+    });
+
+    // Unkown User
```
[the remark this answers is not in the document]
Comment (designxsteph) [reply chocolatey/Website/Scripts/packages/package-details.js]: Typo has been fixed here https://github.com/chocolatey/chocolatey.org/pull/611/commits/8009d99c4ad200fa2fea5eb8fcf14da60c2ace17
